diff --git a/drivers/bluetooth/hci_bcm.c b/drivers/bluetooth/hci_bcm.c
index 32527bdf4b50..fe083e135061 100644
--- a/drivers/bluetooth/hci_bcm.c
+++ b/drivers/bluetooth/hci_bcm.c
@@ -844,7 +844,11 @@ static int bcm_probe(struct platform_device *pdev)
 
 	dev->pdev = pdev;
 
+#if LINUX_VERSION_IS_LESS(4,1,0)
+	if (ACPI_COMPANION(&pdev->dev))
+#else
 	if (has_acpi_companion(&pdev->dev))
+#endif
 		ret = bcm_acpi_probe(dev);
 	else
 		ret = bcm_platform_probe(dev);
